src/components/pagination/pagination.scss

```diff
@@ -0,0 +1,265 @@
+ */
+
+.c-pagination__action.is-previous::before {
+  background-image: svg-load(
+    'icons/arrow-left.svg',
+    color=colors.$primary-brand
+  );
+  left: 0;
+
+  .t-dark & {
+    background-image: svg-load(
+      'icons/arrow-left.svg',
+      color=colors.$text-light-emphasis
+    );
+  }
+}
+
+.c-pagination__action.is-next::after {
+  background-image: svg-load(
+    'icons/arrow-right.svg',
+    color=colors.$primary-brand
+  );
+  right: 0;
+
+  .t-dark & {
+    background-image: svg-load(
+      'icons/arrow-right.svg',
+      color=colors.$text-light-emphasis
+    );
+  }
+}
+
+/**
+ * The inner number element gives us a means of positioning indicator pseudo
+ * elements relative to the action content without disrupting the "previous"
+ * and "next" affordances.
+ *
+ * 1. Constrain the focus ring (not visible initially) to a circle slightly
+ *    smaller than the "previous" and "next" affordances.
+ * 2. Fixes an issue where Chrome renders the digits a pixel or two higher
+ *    than in other browsers.
+ * 3. The aforementioned means of positioning the indicator pseudo element.
+ */
+
+.c-pagination__number {
+  align-items: center;
+  border-radius: sizes.$radius-full; /* 1 */
+  display: flex;
+  font-weight: font.$weight-medium;
+  height: ($action-size - $action-pad * 2); /* 1 */
+  justify-content: center;
+  line-height: ms.step(0, 1); /* 2 */
+  max-width: ($action-size - $action-pad * 2); /* 1 */
+  position: relative; /* 3 */
+  width: 100%; /* 1 */
+}
+
+/**
+ * When keyboard navigation is detected on the containing action, apply a
+ * focus ring similar to other components to the number element.
+ */
+
+.c-pagination__action {
+  @include focus.focus-visible {
+    .c-pagination__number {
+      box-shadow: 0 0 0 sizes.$edge-large colors.$primary-brand-lighter;
+    }
+  }
+}
+
+/**
+ * Hover and current indicators
+ *
+ * 1. Might seem odd compared to `bottom: 0`, but this places the affordance
+ *    within the action padding, which feels nicely balanced.
+ */
+
+.c-pagination__number::after {
+  background: currentColor;
+  border-radius: sizes.$radius-full;
+  content: '';
+  height: $indicator-height;
+  left: 0;
+  margin: auto;
+  position: absolute;
+  right: 0;
+  top: 100%; /* 1 */
+  transition-duration: motion.$speed-quick;
+  transition-property: opacity, transform;
+  transition-timing-function: motion.$ease-out;
+  width: $indicator-width;
+
+  /**
+   * Unless the containing action is the current page or is being hovered,
+   * set this element to its initial animation state.
+   */
+
+  .c-pagination__action:not(:hover):not([aria-current]) & {
+    opacity: 0;
+    transform: scaleX(
+      unit.strip($indicator-height) / unit.strip($indicator-width)
+    );
+  }
+}
+
+/**
+ * Apply unique dimensions to the current state so it feels more like a
+ * pagination dot, as seen in many app interfaces.
+ */
+
+.c-pagination__action[aria-current] .c-pagination__number::after {
```

Comment:
I find the nested Sass parent selector easier to read for situations like this, because it helps me understand that I'm still dealing with the same element, just in a new context. eg:

```scss
.c-pagination__number::after {
  // indicator styles

  .c-pagination__action[aria-current] & {
    // current state modifier
  }
}
```

Reply:
- [x] Moved selector for consistency with previous `:not` selector